controllers/datafoundation_client.go

```diff
@@ -0,0 +1,170 @@
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	ocsClientOperatorName = "ocs-client-operator"
+)
+
+type dataFoundationClientSpec struct {
+	onboardingTicket          string
+	dfstorageProviderEndpoint string
+}
+
+type dataFoundationClientReconciler struct {
+	*ManagedFusionOfferingReconciler
+
+	offering                 *v1alpha1.ManagedFusionOffering
+	dataFoundationClientSpec dataFoundationClientSpec
+	storageclient            ocsclient.StorageClient
+}
+
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclients,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=operators.coreos.com,namespace=system,resources=clusterserviceversions,verbs=get;list;watch;delete;update;patch
+
+func dfcSetupWatches(controllerBuilder *builder.Builder) {
+	csvPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(res client.Object) bool {
+				return strings.HasPrefix(res.GetName(), ocsClientOperatorName)
+			},
+		),
+	)
+	controllerBuilder.
+		Owns(&corev1.Secret{}).
+		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates).
+		Owns(&ocsclient.StorageClient{})
+}
+
+func DFCAddToScheme(scheme *runtime.Scheme) {
+	utilruntime.Must(ocsclient.AddToScheme(scheme))
+}
+
+func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	r := dataFoundationClientReconciler{}
+	r.initReconciler(offeringReconciler, offering)
+
+	if err := r.parseSpec(offering); err != nil {
+		return ctrl.Result{}, err
+	}
+	return r.reconcilePhases()
+}
+
+func (r *dataFoundationClientReconciler) initReconciler(reconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) {
+	r.ManagedFusionOfferingReconciler = reconciler
+	r.offering = offering
+
+	r.storageclient.Name = "ocs-storageclient"
+	r.storageclient.Namespace = offering.Namespace
+}
+
+func (r *dataFoundationClientReconciler) parseSpec(offering *v1alpha1.ManagedFusionOffering) error {
+	r.Log.Info("Parsing ManagedFusionOffering Data Foundation Client spec")
+
+	isValid := true
+	onboardingTicket, found := offering.Spec.Config["onboarding_ticket"]
+	if !found {
+		r.Log.Error(
+			fmt.Errorf("missing field: onboarding_ticket"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	} else if onboardingTicket == "" {
+		r.Log.Error(
+			fmt.Errorf("onboarding_ticket is empty "),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
+
+	dfstorageProviderEndpoint, found := offering.Spec.Config["data_foundation_provider_endpoint"]
```

Comment:
```suggestion
	dfStorageProviderEndpoint, found := offering.Spec.Config["endpoint"]
```
I think the `endpoint` is enough because the endpoint is a specific detail related to DFC offering only. If you think we should have complete `data_foundation_provider_endpoint` then please make it camel-cased

Reply:
I am using `dataFoundationProviderEndpoint` for the key instead of `endpoint`